### ISSUE TITLE
The invoice_ref column of the order table is never populated

### ISSUE THREAD

**Issue by roadster31** [thelia/thelia#2346]
This invoice_ref column is not used, and the order reference is used as the invoice reference.

The $INVOICE_REF output on the order loop is always null.

We should at least populate invoice_ref with the order reference when the order is paid, and use $INVOICE_REF in the invoices instead of $REF.

**Comment by InformatiqueProg** [comment]
This invoice_ref column is not used in Thelia core but it's used in module InvoiceRef (and others ?)
https://github.com/thelia-modules/InvoiceRef

In the same subject $REF and $INVOICE_REF, see PDF INVOICE : Wrong title for $REF, add line for $INVOICE_REF #2242

**Comment by roadster31** [comment]
True, but $INVOICE_REF is not currently populated (e.g. is always NULL), thus we can't use it in the PDF invoice.

**Comment by roadster31** [comment]
@InformatiqueProg , maybe you can add to #2242 an update of the invoice_ref column when the order is paid in Thelia\Action\Order ? It will solve the problem.

But we will have to change the InvoiceRef module event listener priority, which is currently 100, to something higher than 128

**Comment by InformatiqueProg** [comment]
Yes i can ! 😄 
But another day

**Comment by roadster31** [comment]
Sure, this is just a suggestion.

**Comment by InformatiqueProg** [comment]
@roadster31 , i agree with you the $INVOICE_REF would be populate when the order is paid.
But if we use the order $REF for $INVOICE_REF, the $INVOICE_REF numbers will not be continuous, this is a problem in France (and other country ?)

What's the better solution:
a) Integrate the function in module InvoiceRef in the core
b) Your suggestion
c) another idea ?

What's your opinion and other members ?

**Comment by InformatiqueProg** [comment]
Added to PR #2242
Module InvoiceRef PR
https://github.com/thelia-modules/InvoiceRef/pull/1

**Comment by roadster31** [comment]
In Thelia 1, the invoices ref are sequential numbers. We can use the same principle in T2.

**Comment by anoziere** [comment]
Thelia 3 now implements exactly this natively (Domain\Invoice\InvoiceRefAllocator + AllocateInvoiceRefListener: opt-in, gapless yearly invoice sequence allocated when an order becomes paid, never overwrites an existing ref). For 2.6, the need is covered by the actively maintained thelia-modules/InvoiceRef module already discussed in this thread. Suggest closing.
